File: test-suite-kotlin/src/test/kotlin/io/micronaut/tracing/docs/InMemorySpanExporterFactory.kt
package io.micronaut.tracing.docs

import io.micronaut.context.annotation.Factory
import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter
import io.opentelemetry.sdk.trace.SpanProcessor
import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor
import jakarta.inject.Singleton

/**
 * Exports the finished spans of the tests to an in-memory exporter.
 */
@Factory
class InMemorySpanExporterFactory {

    @Singleton
    fun spanProcessor(spanExporter: InMemorySpanExporter): SpanProcessor = SimpleSpanProcessor.create(spanExporter)

    @Singleton
    fun inMemorySpanExporter(): InMemorySpanExporter = InMemorySpanExporter.create()
}

File: test-suite-python/src/test/python/micronaut/tracing/docs/InMemorySpanExporterFactory.py
from io.opentelemetry.sdk.testing.exporter import InMemorySpanExporter
from io.opentelemetry.sdk.trace import SpanProcessor
from io.opentelemetry.sdk.trace.export import SimpleSpanProcessor
from jakarta.inject import Singleton
from micronaut.context.annotation import Factory


@Factory
class InMemorySpanExporterFactory:
    """Exports the finished spans of the tests to an in-memory exporter."""

    @Singleton
    def span_processor(self, span_exporter: InMemorySpanExporter) -> SpanProcessor:
        return SimpleSpanProcessor.create(span_exporter)

    @Singleton
    def in_memory_span_exporter(self) -> InMemorySpanExporter:
        return InMemorySpanExporter.create()
